tests/test_apdb.py

```diff
@@ -87,6 +91,20 @@ def test_load_forced_sources(self):
         sources = self.apdb.load_sources(limit=2)
         self.assertEqual(len(sources), 2)
 
+    def test_make_flag_exclusion_clause(self):
+
+        clause = self.apdb._make_flag_exclusion_clause(self.apdb.bad_DiaSource_flags)
```

Comment:
I'd probably add a docstring that this is testing the default flag list, since it's not obvious why where 972 comes from otherwise? Would probably be worth adding a test that a warning is raised for `mask=0` and at least one other set of flags. Oh, wait, I see you do that below; maybe note *that* in the docstring?